src/obj.rs

```diff
@@ -363,9 +381,15 @@ impl<'a> Parser<'a> {
     self.parse_str().map(Some)
   }
 
-  fn parse_object_name(&mut self) -> Result<String, ParseError> {
-    try!(self.parse_tag("o"));
-    self.parse_str()
+  fn parse_object_name(&mut self) -> Result<Option<String>, ParseError> {
+    if let Some("o") = sliced(&self.peek()) {
```

Comment:
Plus, I’ll switch to a more coherent return value regarding the discusion below about the option.